docs/general/builtins/scalar-functions.rst

```diff
@@ -933,19 +933,42 @@ Example::
     +-----+
     SELECT 1 row in set (... sec)
 
+Example with interval::
+
+    cr> select extract(hour from INTERVAL '5 days 12 hours 45 minutes') AS hour;
+    +------+
+    | hour |
+    +------+
+    |   12 |
+    +------+
+    SELECT 1 row in set (... sec)
+
 Synopsis::
 
     EXTRACT( field FROM source )
 
 ``field``
-  An identifier or string literal which identifies the part of the timestamp
-  that should be extracted.
+  An identifier or string literal which identifies the part of the timestamp or
+  interval that should be extracted.
 
 ``source``
-  An expression that resolves to a timestamp data type with or without timezone
-  or is castable to timestamp data types. In the case the expression has a
-  different return type but is known to be castable to timestamp an implicit
-  cast will be attempted.
+  An expression that resolves to an interval data type, or a timestamp data type
+  (with or without timezone), or is castable to timestamp data types. In the
+  case the expression has a different return type but is known to be castable
+  to timestamp an implicit cast will be attempted.
```

Comment:
```suggestion
  (with or without timezone), or is castable to a timestamp.
```

The last sentence seems to repeat the "castable to a timestamp", but somehow sounds more complicated?